grudge/trace_pair.py

```diff
@@ -402,11 +444,32 @@ def cross_rank_trace_pairs(
         return [TracePair(BTAG_PARTITION(remote_rank), interior=ary, exterior=ary)
                 for remote_rank in connected_ranks(dcoll)]
 
+    actx = get_container_context_recursively(ary)
+
+    try:
+        from grudge.array_context import MPISingleGridWorkBalancingPytatoArrayContext
+    except ImportError:
```

Comment:
Make a somewhat generic way to recognize distributed actxs. (`isinstance(actx, Distributed...?)`)

Reply:
[45e06b0](https://github.com/inducer/grudge/pull/183/commits/45e06b0209b57b1262d0d618f6fce4801f15a33f)